smac/tae/execute_ta_run.py

```diff
@@ -23,6 +24,17 @@ class to define numbers for status types
     ABORT = 4
     MEMOUT = 5
 
+    def enum_hook(obj):
+        """
+        hook function passed to json-deserializer as "object_hook"
+        """
+        if "__enum__" in obj:
+            # object is marked as enum
+            name, member = obj["__enum__"].split(".")
+            return getattr(globals()[name], member)
```

Comment:
Integrated a simple check for the name of the enum. Should a TypeError be raised if another enum than StatusType is detected?

`if name == "StatusType": return getattr(globals()[name], member)`
`else: return obj`